feature/google/src/main/java/com/revenuecat/purchases/google/findBestOffer.kt

```diff
@@ -0,0 +1,108 @@
+@file:Suppress("MaximumLineLength", "MaxLineLength")
+package com.revenuecat.purchases.google
+
+import com.revenuecat.purchases.models.GoogleSubscriptionOption
+import com.revenuecat.purchases.models.PricingPhase
+import com.revenuecat.purchases.models.RecurrenceMode
+
+fun List<GoogleSubscriptionOption>.findBestOffer(): GoogleSubscriptionOption? {
+    val basePlan = this.firstOrNull { it.isBasePlan } ?: return null
+
+    val validOffers = this
+        .filter { !it.isBasePlan }
+        .filter { !it.tags.contains("rc-ignore-best-offer") }
+
+    return findLongestFreeTrial(validOffers) ?: findBestSavingsOffer(basePlan, validOffers) ?: basePlan
+}
+
+private fun findLongestFreeTrial(offers: List<GoogleSubscriptionOption>): GoogleSubscriptionOption? {
+    return offers.mapNotNull { offer ->
+        offer.pricingPhases.filter { pricingPhase ->
+            pricingPhase.priceAmountMicros == 0L
+        }.map {
+            Pair(offer, parseBillPeriodToDays(it.billingPeriod))
+        }.maxByOrNull { it.second }
+    }.maxByOrNull { it.second }?.first
```

Comment:
This logic is _kind_ of gross right now because its mapping `PricingPhase` to an `GoogleSubscriptionOption`. But this is done because we are essentially looking for the `GoogleSubscriptionOption` with the longest free duration and _not_ the `PricinPhase` 🤷‍♂️